ext/DashBasePlotsExt.jl

```diff
@@ -4,6 +4,15 @@ import Plots
 import DashBase
 
 
-DashBase.to_dash(p::Plots.Plot{Plots.PlotlyBackend}) = (data = Plots.plotly_series(p), layout = Plots.plotly_layout(p))
+function DashBase.to_dash(p::Plots.Plot{Plots.PlotlyBackend})
+    return if haskey(Base.loaded_modules, Base.PkgId(Base.UUID("a03496cd-edff-5a9b-9e67-9cda94a718b5"), "PlotlyBase")) &&
+        haskey(Base.loaded_modules, Base.PkgId(Base.UUID("f2990250-8cf9-495f-b13a-cce12b45703c"), "PlotlyKaleido"))
+        # Note: technically it would be sufficient if PlotlyBase is loaded, but thats how it is currently handled by Plots.jl
+        Plots.plotlybase_syncplot(p)
+    else
+        (data = Plots.plotly_series(p), layout = Plots.plotly_layout(p))
+    end
+end
+DashBase.to_dash(p::Plots.Plot{Plots.PlotlyJSBackend}) = Plots.plotlyjs_syncplot(p)
```

Comment:
- Why do we need these two methods?
- You mention that _how it is currently handled by Plots.jl_, where I can find this code?

Reply:
- > You mention that how it is currently handled by Plots.jl, where I can find this code?

https://github.com/JuliaPlots/Plots.jl/blob/7ba5889213bbe201bf7cb2effe77bf3227dbb0cc/src/backends.jl#L583-L616

- > Why do we need these two methods?
- 
I don't understand these convert `Plots.Plot` objects to the corresponding backend objects, such that you could use them in a Dash application, or do you wonder why we need two of them?

Reply:
in 75adf099f756db28a1dd5d917ad0a070e0f5e27a `ext/DashBasePlotsExt.jl` went from defining one `DashBase.to_dash` to two without providing an explanation, hence my question.

Reply:
Ok, and why do we need special logic for when PlotlyKaleido is loaded?

Reply:
That is because the function `Plots.plotly_syncplot` only gets defined if both `PlotlyBase` and `PlotlyKaleido` are loaded (since in Plots.jl, it is only needed when saving to file).

Reply:
That would also work, but not contain the frame key

Reply:
Got it - thanks!

Now, if I understand correctly the package extension mechanism, we could we set up an 

```
ext/DashBasePlotsPlotlyBasePlotlyKaleidoExt.jl
```

extension instead of the `Base.loaded_modules` dance in `ext/DashBasePlotsExt.jl`. Listing it as

```
[extensions]
DashBasePlotsPlotlyBasePlotlyKaleidoExt = ["Plots", "PlotlyBase", "PlotlyKaleido"]
```

in the Project.toml, this new extension when then be loaded only when `Plots`, `PlotlyBase` **and** `PlotlyKaleido` are loaded.

Referencing Lux.jl here:

- https://github.com/LuxDL/Lux.jl/tree/main/ext
- https://github.com/LuxDL/Lux.jl/blob/2dde6f8c398f37e1f0750608d192b81d22c538ec/Project.toml#L36-L45

Reply:
yeah julia doesn't like that method overwriting though
```
┌ DashBasePlotsPlotlyBasePlotlyKaleidoExt [b37a3e6d-8c8a-5d5e-adac-9b3ad75420c2]
│  WARNING: Method definition to_dash(Plots.Plot{Plots.PlotlyBackend}) in module DashBasePlotsExt at /home/schrist/.julia/dev/DashBase/ext/DashBasePlotsExt.jl:7 overwritten in module DashBasePlotsPlotlyBasePlotlyKaleidoExt at /home/schrist/.julia/dev/DashBase/ext/DashBasePlotsPlotlyBasePlotlyKaleidoExt.jl:6.
│    ** incremental compilation may be fatally broken for this module **
└  
```

Reply:
Thanks for checking! Good to know!